src/integration-test/java/io/aiven/kafka/connect/IntegrationTest.java

```diff
@@ -441,6 +441,80 @@ final void jsonlOutputTest() throws ExecutionException, InterruptedException, IO
         }
     }
 
+    @Test
+    final void jsonOutput() throws ExecutionException, InterruptedException, IOException {
+        final Map<String, String> connectorConfig = awsSpecificConfig(basicConnectorConfig(CONNECTOR_NAME));
+        final String compression = "none";
+        final String contentType = "json";
+        connectorConfig.put("format.output.fields", "key,value");
+        connectorConfig.put("format.output.fields.value.encoding", "none");
+        connectorConfig.put("key.converter", "org.apache.kafka.connect.storage.StringConverter");
+        connectorConfig.put("value.converter", "org.apache.kafka.connect.json.JsonConverter");
+        connectorConfig.put("value.converter.schemas.enable", "false");
+        connectorConfig.put("file.compression.type", compression);
+        connectorConfig.put("format.output.type", contentType);
+        connectRunner.createConnector(connectorConfig);
+
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
```

Comment:
In s3 you could use a generators to generate keys and values:
```final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
        final IndexesToString keyGen = (partition, epoch, currIdx) -> "key-" + currIdx;
        final IndexesToString valueGen = (partition, epoch, currIdx) -> "value-" + currIdx;
        for (final KeyValueMessage msg : new KeyValueGenerator(4, 10, keyGen, valueGen)) {
            sendFutures.add(sendMessageAsync(producer, TEST_TOPIC_0, msg.partition, msg.key, msg.value));
        }```

Reply:
Good point, thanks. I will change to use them.

Reply:
@HelenMel done